src/Domain.LinnApps/PurchaseOrders/PurchaseOrderService.cs

```diff
@@ -1055,8 +1056,8 @@ private void PerformDetailCalculations(PurchaseOrderDetail current, PurchaseOrde
 
             current.DetailTotalCurrency = Math.Round(netTotal + current.VatTotalCurrency.Value, 2);
 
-            if (creating || (updated.OrderUnitPriceCurrency == current.OrderUnitPriceCurrency
-                && updated.OurUnitPriceCurrency != current.OurUnitPriceCurrency))
+            if ((creating || (updated.OrderUnitPriceCurrency == current.OrderUnitPriceCurrency
+                && updated.OurUnitPriceCurrency != current.OurUnitPriceCurrency)) && (documentType != "CO" && documentType != "RO"))
```

Comment:
This is a really confusing if statement to read, so maybe I'm getting this wrong.. but looks like 

if documentType is CO, or RO then this if statement is skipped, and OrderUnitPriceCurency is not updated, right?